File: 374_GuessNumberHigherOrLower.py
# The guess API is already defined for you.
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        left, right = 1, n
        mid = (left+right)>>1
        
        res = guess(mid)
        
        while res != 0:
            mid = (left+right)>>1
            res = guess(mid)
            
            if res == 0: return mid
            elif res == -1: right = mid-1
            else: left = mid+1
        return mid

File: cpp/374_GuessNumberHigherOrLower.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        int left = 1, right = n;
        int mid;
        while (left <= right) {
            mid = left + (right - left) / 2;
            int g = guess(mid);
            
            if (g == 0) return mid;
            else if (g == -1) right = mid - 1;
            else left = mid + 1;
        }
       return mid; 
    }
};
